pkg/servicemesh/federation/status/manager_test.go

```diff
@@ -811,6 +811,8 @@ func TestStatusManager(t *testing.T) {
 			for index, f := range tc.events {
 				t.Logf("processing event %d", index)
 				f(handler)
+				// give it a little time for the status update to propagate
+				time.Sleep(25 * time.Millisecond)
```

Comment:
If you are using testify you could wrap verifications in sth like:

`assert.Eventually(t, func() bool { return /*verificaction status*/ true; }, time.Second, 10*time.Millisecond)`

`25ms` can fail on slower machines or when there's quite some load on the cluster running jobs.